### PR TITLE
More options for password passing

### LINKED ISSUE #73026: Non-interactive password prompting for ansible CLIs
##### SUMMARY
This is largely a feature necessary for ansible-runner and Tower, but as of now, to respond to password prompts, runner uses `pexpect`.  

`pexpect` comes with its own host of issues, like the fact that even after you've matched the prompt and sent the response, it still buffers the output for the remainder of execution and adds extra overhead that is unnecessary.

Investigate enabling a way via named pipe, or some other potential ideas, to supply these passwords, other than prompting, or supplying inventory vars.  Providing vars exposes them, and there may be use cases to provide them in a way that the playbook doesn't necessarily have access to.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
lib/ansible/cli

##### ADDITIONAL INFORMATION

```yaml

```

### PR DESCRIPTION
**Pull request by bcoca**
Add a 'file' option for password passing to ansible for connection and become plugins, including 'executable file' functions.

fixes #73026
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

CLI

### CODE REVIEW

File: lib/ansible/config/base.yml [hunk 1]
```diff
@@ -190,6 +190,15 @@ BECOME_ALLOW_SAME_USER:
   - {key: become_allow_same_user, section: privilege_escalation}
   type: boolean
   yaml: {key: privilege_escalation.become_allow_same_user}
+BECOME_PASSWORD_FILE:
+  name: Become password file
+  default: ~
+  description: 'The password file to use for the become plugin. --become-password-file.'
+  env: [{name: ANSIBLE_CONNECTION_PASSWORD_FILE}]
```
Comment (samdoran) [inline lib/ansible/config/base.yml]: Is this supposed to be `ANSIBLE_BECOME_PASSWORD_FILE`?

Comment (bcoca) [reply lib/ansible/config/base.yml]: yep

File: lib/ansible/cli/__init__.py [hunk 2]
```diff
@@ -492,3 +496,40 @@ def get_host_list(inventory, subset, pattern='all'):
...
+
+            try:
+                p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            except OSError as e:
+                raise AnsibleError("Problem occured when trying to run the password script %s (%s)."
+                                   " If this is not a script, remove the executable bit from the file." % (pwd_file, e))
+
+            stdout, stderr = p.communicate()
+            if p.returncode != 0:
+                raise AnsibleError("The password script %s returned and error (rc=%s): %s" % (pwd_file, p.returncode, stderr))
```
Comment (relrod) [inline lib/ansible/cli/__init__.py]:
```suggestion
                raise AnsibleError("The password script %s returned an error (rc=%s): %s" % (pwd_file, p.returncode, stderr))
```

**Comment by briantist** [comment]
This is great! I'm already a (happy) user of the option for ansible-vault. It must have been documented somewhere otherwise I would never have found it, but the additional note about it is good too.

**Pull request merged**